pkg/kapp/crdupgradesafety/change_validator.go

```diff
@@ -241,6 +241,40 @@ func MinimumPropertiesChangeValidation(diff FieldDiff) (bool, error) {
 	}
 }
 
+// DefaultValueChangeValidation adds a validation check to ensure that
+// default values are not changed in a CRD schema:
+// - No new value can be added as default that did not previously have a
+// default value present
+// - Default value of a field cannot be changed
+// - Existing default value for a field cannot be removed
+// This function returns:
+// - A boolean representation of whether or not the change
+// has been fully handled (i.e. the only change was to a field's default value)
+// - An error if either of the above criteria are not met
+func DefaultValueChangeValidation(diff FieldDiff) (bool, error) {
+	handled := func() bool {
+		diff.Old.Default = &v1.JSON{}
+		diff.New.Default = &v1.JSON{}
+		return reflect.DeepEqual(diff.Old, diff.New)
+	}
+
+	switch {
+	case diff.Old.Default == nil && diff.New.Default != nil:
+		return handled(), fmt.Errorf("new value added as default when previously no default value existed: %+v", diff.New.Default.Raw)
```

Comment:
does using the value from the diff object after calling the `handled()` function in the return result in this returning an empty value?

Reply:
I think that we may have some other places in the code this may be a problem but I will plan to address them as part of #945

Reply:
Yes, good catch! I've fixed for now just for this particular change, will capture it as an item as a part of https://github.com/carvel-dev/kapp/issues/945 for all other instances.

Reply:
Reversed this change in the latest commit

Reply:
I think it probably makes sense to address this as part of this PR to prevent additional work in #945 now that we have identified this as a potential issue.

I would recommend doing something like this:
 ```suggestion
                 newDefault := *diff.New.Default.Raw
		return handled(), fmt.Errorf("new value added as default when previously no default value existed: %+v", newDefault)
```

While this introduces a new, temporary, variable which isn't the most ideal it results in a complete error message. IMO a complete error message is more important than reducing temporary variables because it provides a nicer user experience.

Reply:
@rashmigottipati It won't let me unresolve this thread, so tagging you directly

Reply:
Addressed in [latest commit](https://github.com/carvel-dev/kapp/pull/950/commits/97a258940768df26f84ba5297362f15de661d0da)